internals/daemon/daemon.go

```diff
@@ -773,6 +778,13 @@ func syscallModeReboot(rebootDelay time.Duration) error {
 	return nil
 }
 
+func externalModeReboot(rebootDelay time.Duration) error {
+	if rebootDelay > 0 {
+		time.Sleep(rebootDelay)
+	}
```

Comment:
doing this sleep inside HandleRestart seems wrong, you might have to distinguish the calls from HandleRestart vs doReboot now